sqlbp/client.go

```diff
@@ -0,0 +1,190 @@
+package sqlbp
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"time"
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/jmoiron/sqlx"
+	"github.com/luna-duclos/instrumentedsql"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type tracer struct {
+	prefix string
+}
+
+type span struct {
+	tracer
+	parent opentracing.Span
+	err    error
+}
+
+// NewTracer returns a tracer that will fetch spans using opentracing's SpanFromContext function.
+func NewTracer(prefix string) instrumentedsql.Tracer {
+	return tracer{
+		prefix: prefix,
+	}
+}
+
+// GetSpan returns a span from the context, or creates one.
+func (t tracer) GetSpan(ctx context.Context) instrumentedsql.Span {
+
+	if ctx == nil {
```

Comment:
while technically you can pass in a nil ctx, this is considered the caller's fault, so in reality we never check for it. see: https://go.googlesource.com/go/+/go1.16/src/context/context.go#37